.components/custom-layout/CustomLayout.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import Banner from '../banner';
+import CoverImg from '../CoverImg';
+import Promotion from '../Promotion';
+import PromotionBannerCta from '../PromotionBannerCta';
+import Content from '../Content';
+
+const rgxBlogRoute = /^\/blog\/?$/;
+
+const CustomLayout = (props) => {
+  const { headerComp: Header, sidepanelComp: Sidepanel, footerComp: Footer } = props;
+  const isBlogRoute = rgxBlogRoute.test(window.location.pathname);
+  
+  return (
+    <>
+      <Header />
+      <Sidepanel />
+      {(props.frontmatter.bannerTitle || props.frontmatter.title) && (
+        <Banner config={props.config} height={isBlogRoute && '60vh'}>
+          <div style={isBlogRoute ? { maxWidth: '546px', padding: '0 16px' } : { maxWidth: '768px' }}>
+            {props.frontmatter?.bannerDate || props.frontmatter?.date}
+            <h1 id='banner-title'>
+              {props.frontmatter?.bannerTitle || props.frontmatter?.title}
+            </h1>
+            {props.frontmatter?.bannerSubtitle || props.frontmatter?.subtitle}
+          </div>
+          {(props.frontmatter?.bannerImg || props.frontmatter?.ogImg) && (
+            <CoverImg src={props.frontmatter?.bannerImg || props.frontmatter?.ogImg} />
```

Comment:
as this has so many or operations it is difficult to read
create variables like

const showBanner:  boolean = Boolean(props.frontmatter.bannerTitle || props.frontmatter.title)
const bannerTitle = props.frontmatter.bannerTitle || props.frontmatter.title
const bannerData = props.frontmatter?.bannerDate || props.frontmatter?.date
and so on for both Banner and Promotion